### DISCUSSION TITLE
Change log not working for interfaces after Netbox upgrade from 3.3.4 to 3.4.10

### DISCUSSION THREAD

**Discussion by akshay225063** [netbox-community/netbox#13302] [Help Wanted!]
Hello everyone,

I had enable change log for all the interfaces and it was working as expected in 3.3.4 but in 3.4.10 there are changes in changelogging.py file mentioned below:

-  from netbox import thread_locals
+ from netbox.config import _thread_locals

- thread_locals.request = request
- thread_locals.webhook_queue = []
+ _thread_locals.request = request
+ _thread_locals.webhook_queue = []

Do i need to enable or include ,ChangeLoggedModel  in Interface class in dcim/models/device_components.py like below:
class Interface(ModularComponentModel, BaseInterface, CabledObjectModel, PathEndpoint):

I have included code for snapshot and use change_logging:
        if hasattr(netbox_prefix, "snapshot"):
            netbox_prefix.snapshot()

        with self.change_logging(user):
            netbox_prefix.full_clean()
            netbox_prefix.save()

Please suggest if i am missing anything in the process and let me know if anything required to get more clarity on usecase

**Comment by akshay225063** [comment]
Hello ,Can anyone provide any suggestion ?

**Comment by candlerb** [comment]
(Aside: please edit your post and add three backticks (` ``` `) on a line of their own, before and after each code block, so that it formats correctly)

There is no file called "changelogging.py" in Netbox.

There is `netbox/extras/models/change_logging.py`, but it doesn't include the line `from netbox import thread_locals` (not in v3.3.4, not in v3.4.10, and not in v3.5.7)

in v3.3.4 the statement `from netbox import thread_locals` appears in `netbox/netbox/request_context.py`, `netbox/extras/signals.py` and `netbox/extras/context_managers.py`

Therefore, it's not clear what you've done. It *sounds* rather like you're talking about modifying Netbox code itself, in which case, there's no support - you're on your own (you've created your own private fork). If you want to extend Netbox, use the official plugins API.

> I have included code for snapshot and use change_logging:

Where have you included this code? Is this in a custom script or plugin that you're writing? Or somewhere else?

**Comment by akshay225063** [comment]
Thanks for the reply . Currently I am using inspector for interfaces  **Campus Datacenter Autosync** . If I change any values related to interface manually then it will  reflect in the change log to show which object value has been changed but if I do the same procedure using an inspector **Campus Datacenter Autosync** and change log is not working for this inspector .
**NetBox version**
v3.4.10

**Python version**
3.9

![image](https://github.com/netbox-community/netbox/assets/67414818/7b521d1b-3a90-473c-87ec-8ed23eee4945)

**Expected Behaviour**
The changelog to show that the interface fields value like speed and duplex changes when I run dcim.interface inspector **Campus Datacenter Autosync**

**Observed Behaviour**
The changelog does not show anything related to any changes in the value of fields of device's interfaces.

Please suggest ?

**Reply by candlerb** [reply]
> Currently I am using inspector for interfaces Campus Datacenter Autosync

What is that? Is it a Netbox plugin? What's the project's URL? I tried Googling "netbox Campus Datacenter Autosync" and got nothing useful.

> If I change any values related to interface manually then it will reflect in the change log to show which object value has been changed but if I do the same procedure using an inspector Campus Datacenter Autosync and change log is not working for this inspector

Then you need to talk to the vendor of "Campus Datacenter Autosync", whatever that is. If it works by making direct SQL updates to the Postgres database, it definitely won't do change logging. If it uses the REST API, then probably it should. If it uses the Django ORM, then it has to use it correctly.

You still haven't explained how this "Campus Datacenter Autosync" relates to the python code snippets you posted originally.

At this point, since your problem is so unclear, I am going to drop out of this discussion. If you want to frame your question in a way which invites a more helpful answer, I recommend you read this document: <http://www.catb.org/~esr/faqs/smart-questions.html#intro>
